backend/core/domains/bookingflow/services/booking_session_service.py

```
@@ -212,35 +401,169 @@ def complete_booking(session_id):
             if step.id not in completed_step_ids:
                 raise StepValidationError(f"Required step '{step.name}' is not completed")
         
+        # Validate completion type against payment step configuration
+        payment_step = session.booking_flow.steps.filter(step_type='payment_info').first()
+        if payment_step and hasattr(payment_step, 'payment_config'):
+            payment_config = payment_step.payment_config
+            if completion_type == 'quote' and not payment_config.allow_quote_request:
+                raise StepValidationError("Quote requests are not allowed for this booking flow")
```

Ruff (0.13.3)

Avoid specifying long messages outside the exception class

(TRY003)